grails-app/domain/org/pih/warehouse/order/OrderItem.groovy

```diff
@@ -285,14 +286,23 @@ class OrderItem implements Serializable, Comparable<OrderItem> {
         return quantityInvoicedInStandardUom >= quantity
     }
 
+    Integer getInvoicedQuantity() {
+        return allInvoiceItems
+                ?.findAll { !it?.invoice?.isPrepaymentInvoice }
+                ?.sum { it.quantity } ?: 0
+    }
+
     Integer getQuantityAvailableToInvoice() {
         return canceled
                 ? null
-                : quantityShippedInStandardUom - quantityInvoicedInStandardUom
+                : quantityShippedInStandardUom - invoicedQuantity
     }
 
     Boolean isInvoiceable() {
-        return encumbered && (canceled || quantityAvailableToInvoice > 0)
+        return hasPrepaymentInvoice &&
+                !hasRegularInvoice &&
```

Comment:
Is this `!hasRegularInvoice` correct? What if an order item will be split across multiple shipments and then multiple final invoices (= multiple regular invoices)?

Reply:
btw, for order adjustments it makes sense, but I am not sue if it makes sense for order items

Reply:
yes an order item can be split across multiple shipments, but I'm not sure if we support a shipment item being split across multiple invoices. I think we expect that a shipment item can only be invoiced once. This is because if we want to invoice a quantity 2 but a quantity 4 was shipped, then that indicates 2 qty was returned (or whatever) and the shipment item should then be considered fully invoiced (even though we only invoiced quantity 2 of 4).

Reply:
btw here's an example of what an invoice would look like if we have multiple shipments for each order item

![5580b76c-fef9-4d3d-9b25-7a7770d73d7a](https://github.com/user-attachments/assets/fc193441-4bd7-404f-9c40-4b0383a4f1ba)